pkg/build/api/validation/validation.go

```diff
@@ -36,22 +36,43 @@ func ValidateBuildUpdate(build *buildapi.Build, older *buildapi.Build) fielderro
 	return allErrs
 }
 
+func streamTagKey(ref *kapi.ObjectReference) string {
+	if ref.Namespace != "" {
+		return fmt.Sprintf("%s/%s", ref.Namespace, ref.Name)
+	}
+	return ref.Name
+}
+
 // ValidateBuildConfig tests required fields for a Build.
 func ValidateBuildConfig(config *buildapi.BuildConfig) fielderrors.ValidationErrorList {
 	allErrs := fielderrors.ValidationErrorList{}
 	allErrs = append(allErrs, validation.ValidateObjectMeta(&config.ObjectMeta, true, validation.NameIsDNSSubdomain).Prefix("metadata")...)
 
-	// allow only one ImageChangeTrigger for now
+	// allow only one ImageChangeTrigger without a From reference
 	ictCount := 0
+	fromRefs := map[string]struct{}{}
 	for i, trg := range config.Spec.Triggers {
 		allErrs = append(allErrs, validateTrigger(&trg).PrefixIndex(i).Prefix("triggers")...)
-		if trg.Type == buildapi.ImageChangeBuildTriggerType {
-			if ictCount++; ictCount > 1 {
-				allErrs = append(allErrs, fielderrors.NewFieldInvalid("triggers", config.Spec.Triggers, "only one ImageChange trigger is allowed"))
-				break
+		if trg.Type != buildapi.ImageChangeBuildTriggerType {
+			continue
+		}
+		if trg.ImageChange != nil && trg.ImageChange.From != nil {
+			fromKey := streamTagKey(trg.ImageChange.From)
+			_, exists := fromRefs[fromKey]
+			if exists {
+				allErrs = append(allErrs, fielderrors.NewFieldInvalid("triggers", config.Spec.Triggers, "multiple ImageChange triggers refer to the same image stream tag"))
 			}
+			fromRefs[fromKey] = struct{}{}
+			continue
+		}
+
+		// If a From is not specified, we only allow one trigger
+		if ictCount++; ictCount > 1 {
```

Comment:
and it would cover the case where someone defined two triggers:
1) one that defines the from explicitly as the value used by the buildconfig
2) one that has an empty from

which are actually identical but wouldn't be caught by your current logic.